homeassistant/components/rest/switch.py

```diff
@@ -203,9 +211,16 @@ def is_on(self):
         """Get the latest data from REST API and update the state."""
         websession = async_get_clientsession(hass, self._verify_ssl)
 
+        body_state_t = None
+        if self._body_state is not None:
+            body_state_t = self._body_state.async_render()
+
         with async_timeout.timeout(self._timeout):
             req = await websession.get(
-                self._resource, auth=self._auth, headers=self._headers
+                self._resource,
+                auth=self._auth,
+                data=bytes(body_state_t, "utf-8"),
```

Comment:
This makes no sense. GET Requests have no body. Data is passed as part of the url `?bla=1` etc.

You're also calling `bytes` on `None`, which also means that there are no tests included as that blows up 😉

Reply:
I agree that it doesn't make sense to have a body with a GET request.  All I can say is that this is a poorly implemented REST API from this vendor.

As you pointed out I would expect either hitting the endpoint with our without a parameter in the URL would return something.  Bu as you can here all that is returned is an empty response
```
curl -H "Content-Type: application/json" -X GET http://192.168.10.249/mf
curl: (52) Empty reply from server

curl -H "Content-Type: application/json" -X GET http://192.168.10.249/mf?queryDynamicShadowData=1
curl: (52) Empty reply from server
```

But  if I add data to the body then I get the response I'm looking for.
```
 curl -H "Content-Type: application/json" -d '{"queryDynamicShadowData":1}' -X GET http://192.168.10.249/mf
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   340  100   312  100    28   4172    374 --:--:-- --:--:-- --:--:--  4216
{
    "adaptiveLearning": false,
    "awayModeEnabled": false,
    "clientId": "MF_C893468405A4",
    "decommission": false,
    "factoryReset": false,
    "fanDirection": "forward",
    "fanOn": false,
    "fanSleepTimer": 0,
    "fanSpeed": 6,
    "lightBrightness": 100,
    "lightOn": false,
    "lightSleepTimer": 0,
    "resetRfPairList": false,
    "rfPairModeActive": false,
    "schedule": ""
}
```

In the long term I plan to write a fan and light platform for this device.  But for right now I would at least like  to be able to turn it on and off.  Even if I write a platform it still doesn't fix the fact that this is a poorly implemented REST API.  But that's what you get from IoT vendors.

As far as ```bytes``` on ```None``` I see what you're talking about.  I was throwing a TypeError that I didn't see.  I will update so the DEFAULT_BODY_STATE is just an empty string which is what you would expect in the body of a standard REST API GET request.